File: swimiq_mobile/lib/core/swim_time_utils.dart
/// Swim time helpers ported from the existing Streamlit `app.py` logic.
class SwimTimeUtils {
  /// Converts swim time text into seconds.
  ///
  /// Accepts:
  /// - `35.43`
  /// - `1:24.32`
  /// - `5:31.43`
  static double swimTimeToSeconds(String timeText) {
    final trimmed = timeText.trim();
    if (trimmed.isEmpty) {
      throw FormatException('Time is required.');
    }

    if (trimmed.contains(':')) {
      final parts = trimmed.split(':');
      if (parts.length != 2) {
        throw FormatException('Use M:SS.hh format.');
      }

      final minutes = int.parse(parts[0]);
      final seconds = double.parse(parts[1]);
      return double.parse(
        ((minutes * 60) + seconds).toStringAsFixed(2),
      );
    }

    return double.parse(double.parse(trimmed).toStringAsFixed(2));
  }

  /// Converts seconds into swim-time display text.
  static String secondsToSwimTime(double seconds) {
    final minutes = seconds ~/ 60;
    final remaining = seconds % 60;

    if (minutes > 0) {
      final wholeSeconds = remaining.floor();
      final fraction = ((remaining - wholeSeconds) * 100).round();
      return '$minutes:${wholeSeconds.toString().padLeft(2, '0')}.'
          '${fraction.toString().padLeft(2, '0')}';
    }

    return seconds.toStringAsFixed(2);
  }
}

File: android/app/src/main/java/com/swimiq/app/util/SwimTimeUtils.kt
package com.swimiq.app.util

object SwimTimeUtils {
    fun toSeconds(timeText: String): Double {
        val trimmed = timeText.trim()
        if (trimmed.isEmpty()) throw IllegalArgumentException("Time is required.")

        return if (trimmed.contains(":")) {
            val parts = trimmed.split(":")
            if (parts.size != 2) throw IllegalArgumentException("Use M:SS.hh format.")
            val minutes = parts[0].toInt()
            val seconds = parts[1].toDouble()
            ((minutes * 60) + seconds).let { (it * 100).toInt() / 100.0 }
        } else {
            trimmed.toDouble().let { (it * 100).toInt() / 100.0 }
        }
    }

    fun formatSeconds(seconds: Double): String {
        val minutes = (seconds / 60).toInt()
        val remaining = seconds % 60
        return if (minutes > 0) {
            String.format("%d:%05.2f", minutes, remaining)
        } else {
            String.format("%.2f", seconds)
        }
    }
}
